aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -311,13 +314,29 @@ struct ReduceOp {
     }
 
     auto out = (out_scalar_t*)((char*)dst + base_offsets[0]);
+    arg_t* acc = nullptr;
+    if (buf != nullptr) {
+      acc = (arg_t*)((char*)buf + (int)(base_offsets[0] * acc_buffer_multiplier));
```

Comment:
how does this work with acc_buffer_multiplier being possibly non-integer?

Reply:
This should be fine: as `base_offsets[0]` is offset by bytes, given our data structure is assumed to be aligned, the end result would always be an integer.

Reply:
I don't think it is, correct me if I'm wrong: base_offsets[0] will be promoted to float, multiplied with acc_buffer_multiplier, the result will be float and might be nonsensical. If result is larger than 2^24, not all integers are representable, and if you happen to hit one of those, then the result will be rounded.

Reply:
I stand corrected.

Luckily this is host code, so size_t should be perfectly safe. I can have a numerator/denominator.
To avoid overflow, I might need to reduce the fraction as well...

Reply:
This one is not host code :( But it's not in a loop. We should be good.